File: CS/WindowsApplication1/WindowsApplication1/MyPopupMenu.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using DevExpress.Utils.Menu;
using DevExpress.XtraEditors;

namespace WindowsApplication1
{
    public class MyPopupControl : IDXDropDownControl
    {

        public MyPopupControl(Control customControl)
        {
            InitPopupForm(customControl);
        }

        private void InitPopupForm(Control customControl)
        {
            _Form = new Form();
            _Form.FormBorderStyle = FormBorderStyle.None;
            _Form.BackColor = Color.Black;
            _Form.Deactivate += new EventHandler(_Form_Deactivate);
            GroupControl gc = new GroupControl();
            gc.Dock = DockStyle.Fill;
            gc.Text = "Custom popup window";
            _Form.Controls.Add(gc);
            customControl.Dock = DockStyle.Fill;
            gc.Controls.Add(customControl);
        }

        void _Form_Deactivate(object sender, EventArgs e)
        {
            _Form.Hide();
        }


        private Form _Form;
        public Form Form
        {
            get { return _Form; }
            set { _Form = value; }
        }


        #region IDXDropDownControl Members


        void IDXDropDownControl.Hide()
        {
            Form.Hide();
        }

        void IDXDropDownControl.Show(IDXMenuManager manager, Control control, Point pos)
        {
            Form.Show();
            Form.Location = control.PointToScreen(pos);
        }

        bool IDXDropDownControl.Visible
        {
            get { return Form.Visible; }
        }

        #endregion
    }
}

File: VB/WindowsApplication1/MyPopupMenu.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms
Imports DevExpress.Utils.Menu
Imports DevExpress.XtraEditors

Namespace WindowsApplication1
	Public Class MyPopupControl
		Implements IDXDropDownControl

		Public Sub New(ByVal customControl As Control)
			InitPopupForm(customControl)
		End Sub

		Private Sub InitPopupForm(ByVal customControl As Control)
			_Form = New Form()
			_Form.FormBorderStyle = FormBorderStyle.None
			_Form.BackColor = Color.Black
			AddHandler _Form.Deactivate, AddressOf _Form_Deactivate
			Dim gc As New GroupControl()
			gc.Dock = DockStyle.Fill
			gc.Text = "Custom popup window"
			_Form.Controls.Add(gc)
			customControl.Dock = DockStyle.Fill
			gc.Controls.Add(customControl)
		End Sub

		Private Sub _Form_Deactivate(ByVal sender As Object, ByVal e As EventArgs)
			_Form.Hide()
		End Sub


		Private _Form As Form
		Public Property Form() As Form
			Get
				Return _Form
			End Get
			Set(ByVal value As Form)
				_Form = value
			End Set
		End Property


		#Region "IDXDropDownControl Members"


		Private Sub Hide() Implements IDXDropDownControl.Hide
			Form.Hide()
		End Sub

		Private Sub Show(ByVal manager As IDXMenuManager, ByVal control As Control, ByVal pos As Point) Implements IDXDropDownControl.Show
			Form.Show()
			Form.Location = control.PointToScreen(pos)
		End Sub

		Private ReadOnly Property Visible() As Boolean Implements IDXDropDownControl.Visible
			Get
				Return Form.Visible
			End Get
		End Property

		#End Region
	End Class
End Namespace
